autogen/oai/client.py

```diff
@@ -49,6 +51,209 @@
     logger.addHandler(_ch)
 
 
+class Client(ABC):
+    """
+    A client class must implement the following methods:
+    - create must return a response object that implements the ClientResponseProtocol
+    - cost
+
+    This class is used to create a client that can be used by OpenAIWrapper.
+    It mimics the OpenAI class, but allows for custom clients to be used.
+    """
+
+    RESPONSE_USAGE_KEYS = ["prompt_tokens", "completion_tokens", "total_tokens", "cost", "model"]
+
+    class ClientResponseProtocol(Protocol):
+        class Choice(Protocol):
+            class Message(Protocol):
+                content: str | None
+                function_call: str | None
```

Comment:
`tool_calls` is now part of the openai schema.

Reply:
Here I have included the minimum elements that are called in the client code.

If we want a mode complete solution, we could change this Protocol to just follow the openai `ChatCompletions` response completely.

Reply:
Fair point. I think in the short term OpenAI's message format is what we are going to be using.

Reply:
@olgavrou shall we switch to use `ChatCompletion` response?